synapse/storage/controllers/state.py

```diff
@@ -542,7 +554,7 @@ async def get_current_hosts_in_room(self, room_id: str) -> List[str]:
 
         await self._partial_state_room_tracker.await_full_state(room_id)
 
-        return await self.stores.main.get_current_hosts_in_room(room_id)
+        return await self.stores.main.get_current_hosts_in_room_ordered(room_id)
 
     async def get_current_hosts_in_room_or_partial_state_approximation(
```

Comment:
We could probably simplify the implementation of `get_current_hosts_in_room_or_partial_state_approximation`.
I intended it to match the behaviour of `get_current_hosts_in_room` for full state rooms, except there's no longer any ordering.